src/inc/volatile.h

```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__BIONIC__))
```

Comment:
What was the problem you got with the `__has_builtin`?

`__atomic_load` is compiler intrinsic, so I would think that we should be checking compiler version whether it is ok to use it. The presence of the intrinsic should not depend on the C runtime used.

cc @janvorli